sycl/test/filter_selector/select.cpp

```diff
@@ -0,0 +1,133 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t1.out
+// RUN: %t1.out
+
+//==------------------- select.cpp - string_selector test ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+using namespace cl::sycl::ONEAPI;
+
+int main() {
+  std::vector<device> CPUs;
+  std::vector<device> GPUs;
+  std::vector<device> Devs;
+  device host;
+
+  CPUs = device::get_devices(info::device_type::cpu);
+  GPUs = device::get_devices(info::device_type::gpu);
+  Devs = device::get_devices();
+
+  std::cout << "# CPU Devices found: " << CPUs.size() << std::endl;
+  std::cout << "# GPU Devices found: " << GPUs.size() << std::endl;
+  std::cout << "# Devices found: " << Devs.size() << std::endl;
+
+  bool HasLevelZero = false;
+  bool HasOpenCL = false;
+
+  auto Platforms = platform::get_platforms();
+  for (auto Platform : Platforms) {
+    if (!Platform.is_host()) {
+      auto Backend = Platform.get_backend();
+      if (Backend == backend::level_zero) {
+        HasLevelZero = true;
+      } else if (Backend == backend::opencl) {
+        HasOpenCL = true;
+      }
+    }
+  }
+
+  if (!CPUs.empty()) {
+    std::cout << "Test 'cpu'" << std::endl;
+    device d1(filter_selector("cpu"));
+    assert(d1.is_cpu());
+  }
+
+  if (!GPUs.empty()) {
+    std::cout << "Test 'gpu'" << std::endl;
+    device d2(filter_selector("gpu"));
+    assert(d2.is_gpu());
+
+    std::cout << "Test 'cpu,gpu'" << std::endl;
+    device d3(filter_selector("cpu,gpu"));
```

Comment:
we may have GPU without CPU.
So this case should be wrapped with if( !CPUs.empty() && !GPUs.empty())

Reply:
For this test, I don't think so - if there's no CPU, then a GPU should be returned.

Reply:
In this case  it is better to wrap with if( !CPUs.empty() || !GPUs.empty()) to let it work if only CPU RT is present

Reply:
Ok